packages/webclient/src/AuthorizationHandlerFactory.ts

```diff
@@ -1,35 +1,26 @@
-import ex = require('@actions/exec');
-import io = require('@actions/io');
-import AzAuthInitializer = require('./AuthHandler/AzCliAuthHandler');
-
+import * as core from '@actions/core';
 import { AzCliAuthHandler } from "./AuthHandler/AzCliAuthHandler";
 import { AzureEndpoint } from "./AuthHandler/AzureEndpoint";
 import { IAuthorizationHandler } from "./AuthHandler/IAuthorizationHandler";
 import { exists } from "@actions/io/lib/io-util";
 
 export const authFilePath: string = "/home/auth.json"
 
-export async function getHandler(): Promise<IAuthorizationHandler> {
-    let azPath = await io.which("az", true);
-    let stdout = '';
-    let code = await ex.exec(`"${azPath}" account show`, [], { silent: true,
-        listeners: {
-            stdout: (data: Buffer) => {
-                stdout += data.toString();
-            }
-        } 
-    });
-    let resp = JSON.parse(stdout);
-    let fileExist = await exists(authFilePath);
-
-    if(code == 0 && !!resp && !!resp.id) {
-        await AzAuthInitializer.initialize();
-        return AzCliAuthHandler.getEndpoint(resp.id);
-    }
-    else if(fileExist) {
-        return AzureEndpoint.getEndpoint(authFilePath);
-    }
-    else {
+export class AuthorizationHandlerFactory {
+    public static async getHandler(): Promise<IAuthorizationHandler> {
+        try {
+            core.debug('try-get AzCliAuthHandler');

```

Comment:
If login action is present before and endpoint object creation fails we do not want to check for auth.json existence. Removing if has changes the logic itself. Also exists() returns a promise, you need to await the result.